packages/calcite-components/src/components/list/list.tsx

```diff
@@ -742,6 +748,35 @@ export class List implements InteractiveComponent, LoadableComponent, SortableCo
     }
   };
 
+  private findAncestorOfFirstFilteredItem = (filteredItems: HTMLCalciteListItemElement[]): void => {
+    this.ancestorOfFirstFilteredItem?.removeAttribute("data-filter");
+    filteredItems.forEach((item) => {
+      item.removeAttribute("data-filter");
+    });
+
+    this.ancestorOfFirstFilteredItem = this.getTopLevelAncestorItemElement(filteredItems[0]);
+    filteredItems[0].setAttribute("data-filter", "0");
```

Comment:
Totally missed this last time, but you can use `toggleAttribute` if we don't care about the value. Not critical for this PR.

Reply:
i think we can rename the attribute to be something like `data-filtered-first` so that we can leverage `toggleAttribute`.